### PR TITLE
Jump to newest message with refrsh

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Draft for fix of gap created after clicking in bottom to go to first messages.

### 🛠 Implementation details

Allow request to have information about the need of messages to be refreshed.

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/179286964-3e26331d-9c30-417f-8c74-3b5ce5046b65.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/179286997-22df2168-d516-44a8-89fb-f11d27611fa8.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>
### 🧪 Testing

On develop:
- Search for a message 
- Click in the buttom to go to bottom
- Check that a gap in messages are created

Now move to this branch and check that the problem is fixed.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- ~[ ] PR is linked to the GitHub issue it resolves~ This fixes something now published yet.

#### Code & documentation
- ~[ ] Changelog is updated with client-facing changes~.  This fixes something now published yet. 
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt [hunk 1]
```diff
@@ -466,9 +466,8 @@ internal class ChannelStateLogicImpl(
 
         updateDataFromChannel(
             channel,
-            shouldRefreshMessages = request.isFilteringAroundIdMessages() ||
-                (!request.isFilteringMessages() && !mutableState.insideSearch.value),
-            scrollUpdate = request.isFilteringNewerMessages() || request.filteringOlderMessages()
+            shouldRefreshMessages = request.shouldRefresh,
```
Comment (TGazica) [inline stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt:469]:
```suggestion
            shouldRefreshMessages = request.shouldRefresh || !request.isFilteringMessages(),
```

**Review by TGazica — CHANGES_REQUESTED** [review]

**Comment by kanat** [comment]
Is there a ticket for this PR?

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt:469]: Let's then move it to `watch` so we have a simple logic here.

**Comment by leandroBorgesFerreira** [comment]
Linkeed to tickets

**Review by TGazica — APPROVED** [review]
Looks good

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] Bugs validated (bugfixes)
- [x] All code we touched has new or updated KDocs

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**